### PR TITLE
docs: add installation page to main documentation

### LINKED ISSUE #464: Add "Installation" page to main documentation
We should include a guide to installing autora in ones' local environment.
The guide should include:

- [x] List of supported python versions
- [x] Basic pip/conda install instruction
- [x] Introduction to packaging structure optional dependencies, which could build on this: https://autoresearch.github.io/autora/contribute/#project-structure
- [x] Example for optional package installations with pip/conda
- [x] A (curated) list of all available vetted (optional) autora pip/conda packages along with a link to their respective page in the documentation. Perhaps this could be a table?
- [ ] We should also consider adding the name of the autora pip/conda package for each tool in the main doc file (index.md) of the respectvie package.

### PR DESCRIPTION
**Pull request by benwandrew**
# Description

resolves #464 

# Type of change

- **docs**: Documentation only changes

# Questions (Optional)
- not sure where to put this... first page under "User Guide"?
- also not sure what @musslick mean't by the last task -- "We should also consider adding the name of the autora pip/conda package for each tool in the main doc file (index.md) of the respectvie package."

# Remarks (Optional)
- to me, listing all the optional dependencies seems overkill and likely to get lost in the process of adding new ones going forward. as such, i opted to refer people to the `pyproject.toml` file for a complete list of options. this is comprehensive, and will necessarily be updated as optional dependencies are added. happy to change and create some sort of table, but aiming for effective viable solution that doesn't add documentation liability in the future...

### CODE REVIEW

File: docs/installation.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+```
+Python 3.11.3 (main, Apr  7 2023, 20:13:31) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
+Type "help", "copyright", "credits" or "license" for more information.
+```
+## Install `autora`
+
+To install the PyPI `autora` package, run the following command:
+
+```shell
+pip install "autora"
```
Comment (younesStrittmatter) [inline docs/installation.md:35]: should we consistently use ```pip install "autora"``` or ```pip install autora```?

File: docs/installation.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+[`autora`](https://github.com/autoresearch/autora) is the parent package which end users are expected to install (as specified above). The parent depends on core packages, such as [`autora-core`](https://github.com/autoresearch/autora-core), [`autora-workflow`](https://github.com/autoresearch/autora-workflow), and [`autora-synthetic`](https://github.com/autoresearch/autora-synthetic). It also includes vetted modules (child packages) as optional dependencies which users can choose to install.
+
+For a complete list of optional dependencies that have been vetted by the core AutoRA team, see the `[project.optional-dependencies]` section of the `pyproject.toml` file in the parent `autora` package.
+
+## Install Optional Dependencies
+
+To install any (combintation) of optional dependencies, users should run the relevant analogue of the following command, with the name in brackets matching the name as specified in the parent `pyproject.toml` file:
+
+```shell
+pip install -U autora[desired-dependency]
```
Comment (younesStrittmatter) [inline docs/installation.md:63]:
```suggestion
pip install -U "autora[desired-dependency]"
```

File: docs/installation.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+```
+
+For example, to install one of the [Theorists](theorist/index.md), such as the Bayesian Machine Scientist (BMS), a user should run:
+
+```shell
+pip install -U autora[theorist-bms]
```
Comment (younesStrittmatter) [inline docs/installation.md:69]:
```suggestion
pip install -U "autora[theorist-bms]"
```

File: docs/installation.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+```
+
+To check that installation was successful, a user can try importing one of the main classes of the corresponding child package. For BMS, such a check would be:
+```shell
+python -c from autora.theorist.bms import BMSRegressor; BMSRegressor()
```
Comment (younesStrittmatter) [inline docs/installation.md:74]:
```suggestion
python -c "from autora.theorist.bms import BMSRegressor; BMSRegressor()"
```

**Review by younesStrittmatter — CHANGES_REQUESTED** [review]

File: docs/installation.md (hunk 1 above)
Comment (benwandrew) [reply docs/installation.md:35]: good catch, thanks! on that point, i think we still have some cleaning up to do globally... eg, https://autoresearch.github.io/autora/workflow-tutorial/docs/interactive/Basic%20Usage/#:~:text=%23%20!pip%20install%20autora

i'll make an issue.

File: docs/installation.md (hunk 4 above)
Comment (benwandrew) [reply docs/installation.md]: actually, are the quotes accurate in this case? if so, we'll need to update the individual `quickstart.md` pages (eg, https://github.com/AutoResearch/autora-theorist-bms/blob/main/docs/quickstart.md)

**Comment by musslick** [comment]
Hey @benwandrew , regarding the last task:
We should list the name of each optional dependency that can be installed with autora. Ideally in a table, e.g.,

```
| type | package name | description |  depenency |
-------------------------------------------------
| experimentalist | experimentalist-falsification | samples experimental conditions that are predicted to invalidate a model | optional |
```

**Comment by musslick** [comment]
@benwandrew 
Regarding the remark: I think we should still list all packages in the doc. Better an old list than no list at all. Pointing the user to the pytoml is cumbersome for them and will scare off unexperienced users.

File: docs/installation.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+## Project Structure
+
+AutoRA is organized into one "parent" and many "child" packages.
+
+![image](img/package_overview.png)
+
+[`autora`](https://github.com/autoresearch/autora) is the parent package which end users are expected to install (as specified above). The parent depends on core packages, such as [`autora-core`](https://github.com/autoresearch/autora-core), [`autora-workflow`](https://github.com/autoresearch/autora-workflow), and [`autora-synthetic`](https://github.com/autoresearch/autora-synthetic). It also includes vetted modules (child packages) as optional dependencies which users can choose to install.
+
+For a complete list of optional dependencies that have been vetted by the core AutoRA team, see the `[project.optional-dependencies]` section of the `pyproject.toml` file in the parent `autora` package.
```
Comment (musslick) [inline docs/installation.md:56]: I would still add a section with the complete list of **optional** dependencies (see the other comment)

**Review by musslick — CHANGES_REQUESTED** [review]
Looks all great, except we should still add a complete list of dependencies in a table (e.g., type, package name, description; see the example in the comment)

File: docs/installation.md (hunk 5 above)
Comment (younesStrittmatter) [reply docs/installation.md:56]: maybe we should add that to the workflow when adding a new dependency? (update pyproject.toml, mkdocs.yml file + list)?

Comment (musslick) [reply docs/installation.md]: Yes, great point!

Comment (musslick) [reply docs/installation.md]: OK this is now added and resolved.

**Comment by musslick** [comment]
> 

Now resolved!

**Review by musslick — APPROVED** [review]
Added the list of dependencies and restructured a bit. Now looks good on my end!

**Comment by benwandrew** [comment]
ok, all sounds good. and thanks for handling the list! 

assume we're happy with placement as first page under User Guide?

**Comment by musslick** [comment]
> ok, all sounds good. and thanks for handling the list!
> 
> assume we're happy with placement as first page under User Guide?

Yes, very happy :)

**Comment by benwandrew** [comment]
@younesStrittmatter can you approve this when you get a chance? because you requested changes, it is not letting me merge until you approve. or perhaps i'm missing another outstanding requested change?

**Review by younesStrittmatter — APPROVED** [review]
looks good to me

**Comment by benwandrew** [comment]
upon another quick read, wondering why certain samplers are listed as required dependencies? if this needs changing, i might also grab a couple typos in another PR...

Screen Shot 2023-07-03 at 15 08 55

**Pull request merged**